base/applications/osk/main.c

```diff
@@ -178,17 +180,30 @@ int OSK_DlgInitDialog(HWND hDlg)
     monitor = MonitorFromPoint(Pt, MONITOR_DEFAULTTOPRIMARY);
     info.cbSize = sizeof(info);
     GetMonitorInfoW(monitor, &info);
-
-    /* Move the dialog on the bottom of main screen */
     GetWindowRect(hDlg, &rcWindow);
-    MoveWindow(hDlg,
-               (info.rcMonitor.left + info.rcMonitor.right) / 2 - // Center of screen
-                   (rcWindow.right - rcWindow.left) / 2,          // - half size of dialog
-               info.rcMonitor.bottom -               // Bottom of screen
-                   (rcWindow.bottom - rcWindow.top), // - size of window
-               rcWindow.right - rcWindow.left,     // Width
-               rcWindow.bottom - rcWindow.top,     // Height
-               TRUE);
+    GetWindowRect(GetDesktopWindow(), &rcDesktop);
+    rgn = CreateRectRgn(rcDesktop.left, rcDesktop.top, rcDesktop.right, rcDesktop.bottom);
```

Comment:
You are leaking `rgn`
Besides, this is not a good way to check.

You could use for example: `MonitorFromWindow`, with `MONITOR_DEFAULTTONEAREST` to get the monitor with the biggest overlap with the window (or default to the nearest).
Then, using that you can get the rectangle of the monitor with `GetMonitorInfo`,
and calculate the intersection between the window (`rcWindow`) and the monitor workspace (`.rcWork`) using `IntersectRect`.
If there is no intersection, `IntersectRect` will indicate this, and if there is some intersection, you can always check if it's sufficient in the resulting rect.